main/src/main/scala/sbt/Defaults.scala

```diff
@@ -2460,10 +2461,16 @@ object Defaults extends BuildCommon {
         val bspTask = (compile / bspCompileTask).value
         val result = cachedCompileIncrementalTask.result.value
         val reporter = (compile / bspReporter).value
-        val store = analysisStore
+        val store = analysisStore(compileAnalysisFile)
         val ci = (compile / compileInputs).value
+        val c = fileConverter.value
+        val dir = c.toPath(backendOutput.value).toFile
         result match
           case Result.Value(res) =>
+            val rawJarPath = c.toPath(res._2)
+            IO.delete(dir)
+            IO.unzip(rawJarPath.toFile, dir)
+            IO.delete(dir / "META-INF" / "MANIFEST.MF")
```

Comment:
I am not a huge fan of this but we need to sync the class folder and the cached jar if in order to have incremental compilation.

Alternatively we could cache the folder instead of the jar, and the cache would take care of syncing it when needed. I would prefer this solution.